tex/chProgramming.tex

```diff
@@ -962,6 +973,25 @@ \subsection{Recursion on natural numbers}\label{ssec:recnat}
 what the \C{(addn n m)} program does is simply to pile \C{n} times the
 successor symbol on top of \C{m}.
 
+If we reflect again on the discussion of Peano arithmetics, as in
+Section~\ref{sec:data}, we see that addition is provided by the definition of
+\C{addn}, and the usual axioms of Peano arithmetic, namely:
+\[S~ x + y = S (x + y) \qquad x + 0 = x \]
```

Comment:
This looks strange. Why are you destructing the first argument in the first equality and the second argument in the second? Shouldn't it either be (S x) + y = S (x + y) and 0 + y = y, or x + (S y) = S (x + y) and x + 0 = x ? (I'd also put a comma between the two equations.)